src/Microsoft.Azure.WebJobs.Host/Blobs/Triggers/BlobTriggerBinding.cs

```diff
@@ -190,8 +190,11 @@ private FileAccess Access
 
         private static IReadOnlyDictionary<string, Type> CreateBindingDataContract(IBlobPathSource path)
         {
-            Dictionary<string, Type> contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+            var contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
             contract.Add("BlobTrigger", typeof(string));
+            contract.Add("Uri", typeof(Uri));
+            contract.Add("Properties", typeof(BlobProperties));
```

Comment:
This BlobProperties object has a bunch of useful info on it, Content-Type, etc. This object is useful in core SDK scenarios where the function might be binding to a Stream for example.

The challenge will be making this available for example to Node.js functions in Azure Functions. I didn't want to cherry pick properties and hoist them as top level. I think what I'll do is add the capability for simple property bag objects like this to be translated into a dictionary, so for a Node function, it would be accessible as a regular object. Those changes will be in Functions.